posydon/binary_evol/SN/step_SN.py

```diff
@@ -927,6 +930,55 @@ def PISN_prescription(self, star):
                         m_PISN = m_star
                     else:
                         m_PISN = m_He_core
+            
+            elif self.PISN == 'Hendriks+23':
+                # Hendriks et al. 2023 PISN prescription
+                # Shifting PPI and PISN gap
+                # works by removing delta_M_PPI from the star
+                # and then applying any remnant mass prescription
+                
+                delta_M_CO_shift = self.PISN_CO_shift if self.PISN_CO_shift is not None else 0.0
+                delta_M_PPI_extra_ML = self.PPI_extra_mass_loss if self.PPI_extra_mass_loss is not None else 0.0
+                if ((m_CO_core >= 38 + delta_M_CO_shift) 
+                    and m_CO_core <= 114 + delta_M_CO_shift):
+
+                    delta_M_PPI = (
+                        (0.0006 * np.log10(star.metallicity * const.Zsun) + 0.0054)
+                        * (m_CO_core - delta_M_CO_shift - 34.8)**3
+                        - 0.0013 * (m_CO_core - delta_M_CO_shift - 34.8)**2
+                        + delta_M_PPI_extra_ML
+                    )
+                    if self.verbose:
+                        print(f"delta_M_PPI: {delta_M_PPI} Msun")
+                else:
+                    delta_M_PPI = 0.0
+                    
+                if delta_M_PPI <= 0.0:
+                    if self.conserve_hydrogen_envelope:
+                        m_PISN = m_star
+                    else:
+                        m_PISN = m_He_core
+                else:
+                    if self.conserve_hydrogen_envelope:
+                        m_PISN = m_star - delta_M_PPI
+                    else:
+                        m_PISN = m_He_core - delta_M_PPI
+                    
+                    if m_PISN < 0.0:
+                        m_PISN = np.nan
```

Comment:
Isn't this the case, where the PI leaves no remnant? Hence, shouldn't the following calculation of `m_rembar` only be done otherwise (having it in an else case)?
At the moment the calculation of `m_rembar` can overwrite the full disruption of the star.